File: Estructurales Java/Bridge.java
// Bridge Pattern
interface Implementation {
    String operationImplementation();
}

class ConcreteImplementationA implements Implementation {
    @Override
    public String operationImplementation() {
        return "ConcreteImplementationA: Here's the result on platform A.";
    }
}

class ConcreteImplementationB implements Implementation {
    @Override
    public String operationImplementation() {
        return "ConcreteImplementationB: Here's the result on platform B.";
    }
}

class Abstraction {
    protected Implementation implementation;
    
    public Abstraction(Implementation implementation) {
        this.implementation = implementation;
    }
    
    public String operation() {
        return "Abstraction: Base operation with:\n" + implementation.operationImplementation();
    }
}

class ExtendedAbstraction extends Abstraction {
    public ExtendedAbstraction(Implementation implementation) {
        super(implementation);
    }
    
    @Override
    public String operation() {
        return "ExtendedAbstraction: Extended operation with:\n" + implementation.operationImplementation();
    }
}

// Usage
class BridgeDemo {
    public static void main(String[] args) {
        Implementation implementationA = new ConcreteImplementationA();
        Abstraction abstraction = new Abstraction(implementationA);
        System.out.println(abstraction.operation());
        
        Implementation implementationB = new ConcreteImplementationB();
        ExtendedAbstraction extended = new ExtendedAbstraction(implementationB);
        System.out.println(extended.operation());
    }
}

File: Estructurales Python/bridge.py
# Bridge Pattern
from abc import ABC, abstractmethod

class Implementation(ABC):
    @abstractmethod
    def operation_implementation(self):
        pass

class ConcreteImplementationA(Implementation):
    def operation_implementation(self):
        return "ConcreteImplementationA: Here's the result on platform A."

class ConcreteImplementationB(Implementation):
    def operation_implementation(self):
        return "ConcreteImplementationB: Here's the result on platform B."

class Abstraction:
    def __init__(self, implementation: Implementation):
        self.implementation = implementation
    
    def operation(self):
        return f"Abstraction: Base operation with:\n{self.implementation.operation_implementation()}"

class ExtendedAbstraction(Abstraction):
    def operation(self):
        return f"ExtendedAbstraction: Extended operation with:\n{self.implementation.operation_implementation()}"

# Usage
if __name__ == "__main__":
    implementation_a = ConcreteImplementationA()
    abstraction = Abstraction(implementation_a)
    print(abstraction.operation())
    
    implementation_b = ConcreteImplementationB()
    extended = ExtendedAbstraction(implementation_b)
    print(extended.operation())
